tests/Export/Adapters/AttributeAdapterTest.php

```diff
@@ -0,0 +1,830 @@
+    use ConfigHelper;
+
+    /** @var SalesChannelContext */
+    protected $salesChannelContext;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->salesChannelContext = $this->buildSalesChannelContext();
+        $this->getContainer()->set('fin_search.sales_channel_context', $this->salesChannelContext);
+        DynamicProductGroupService::getInstance(
+            $this->getContainer(),
+            $this->getContainer()->get('serializer.mapping.cache.symfony'),
+            Context::createDefaultContext(),
+            'ABCDABCDABCDABCDABCDABCDABCDABCD',
+            0
+        );
+        $this->getContainer()->set(
+            'fin_search.export_context',
+            new ExportContext(
+                'ABCDABCDABCDABCDABCDABCDABCDABCD',
+                [],
+                $this->getCategory()
+            )
+        );
+    }
+
+    public function testAttributeContainsAttributeOfTheProduct(): void
+    {
+        $adapter = $this->getContainer()->get(AttributeAdapter::class);
+        $product = $this->createTestProduct();
+        $expected = $this->getAttributes($product, IntegrationType::API);
+
+        $attribute = $adapter->adapt($product);
+
+        $this->assertEquals($expected, $attribute);
+    }
+
+    /**
+     * @dataProvider attributeProvider
+     */
+    public function testAttributesAreProperlyEscaped(
+        string $integrationType,
+        string $attributeName,
+        string $expectedName
+    ): void {
+        $config = $this->getMockedConfig($integrationType);
+
+        $adapter = $this->getAttributeAdapter($config);
+
+        $productEntity = $this->createTestProduct(
+            [
+                'properties' => [
+                    [
+                        'id' => Uuid::randomHex(),
+                        'name' => 'some value',
+                        'group' => [
+                            'id' => Uuid::randomHex(),
+                            'name' => $attributeName
+                        ],
+                    ]
+                ]
+            ]
+        );
+
+        $attributes = $adapter->adapt($productEntity);
+
+        $foundAttributes = array_filter(
+            $attributes,
+            static function (Attribute $attribute) use ($expectedName) {
+                return $attribute->getKey() === $expectedName;
+            }
+        );
+
+        /** @var Attribute $attribute */
+        $attribute = reset($foundAttributes);
+        $this->assertInstanceOf(
+            Attribute::class,
+            $attribute,
+            sprintf('Attribute "%s" not present in attributes.', $expectedName)
+        );
+    }
+
+    /**
+     * @dataProvider multiSelectCustomFieldsProvider
+     */
+    public function testProductWithMultiSelectCustomFields(
+        array $customFields,
+        array $expectedCustomFieldAttributes
+    ): void {
+        $data['customFields'] = ['long_value' => str_repeat('und wieder, ', 20000)];
```

Comment:
`$customFields` is never used, did you mean?:
```suggestion
    public function testProductWithMultiSelectCustomFields(
        array $customFields,
        array $expectedCustomFieldAttributes
    ): void {
        $data['customFields'] = $customFields;
```